internal/crypto/crypto.go

```diff
@@ -233,91 +235,142 @@ func (k *EncryptionKey) Valid() bool {
 // holds the plaintext.
 var ErrInvalidCiphertext = errors.New("invalid ciphertext, same slice used for plaintext")
 
-// Encrypt encrypts and authenticates data. Stored in ciphertext is IV || Ciphertext ||
-// MAC. Encrypt returns the new ciphertext slice, which is extended when
-// necessary. ciphertext and plaintext may not point to (exactly) the same
-// slice or non-intersecting slices.
-func (k *Key) Encrypt(ciphertext []byte, plaintext []byte) ([]byte, error) {
+// validNonce checks that nonce is not all zero.
+func validNonce(nonce []byte) bool {
+	sum := 0
+	for b := range nonce {
+		sum += b
```

Comment:
This might be faster with `|=` and `var sum byte`.

Reply:
You're right, it's ever so slightly faster, at least on amd64:
```
$ benchcmp bench-int bench-byte
benchmark                 old ns/op     new ns/op     delta
BenchmarkNonceValid-4     9.45          8.22          -13.02%
```

But, much more importantly: After I made `sum` a `byte`, the compiler suddenly complained that `b` had the wrong type: `int` instead of `byte`. Turns out, the function has a very subtle bug: It didn't sum up the elements of `nonce`, but the indexes. :disappointed: 

So, thanks again for the review, it helped find a bug! (And I've also added tests for thatt).

Reply:
Woo, good catch!